app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -3946,6 +3952,31 @@ class BrowserTabViewModel @Inject constructor(
         newTabPixels.get().fireNewTabDisplayed()
     }
 
+    private suspend fun shouldPromptForBrokenSiteReport(): Boolean {
+        return withContext(dispatchers.io()) {
+            val refreshCount = brokenSitePrompt.getUserRefreshesCount()
+            when (refreshCount) {
+                0, 1 -> false
+                2 -> {
+                    blockListPixelsPlugin.get2XRefresh()?.getPixelDefinitions()?.forEach {
```

Comment:
wondering whether we should just send the number of refreshes to RefreshPixelSender and keep firing the pixels there

Reply:
+1, I would generally remove as much logic from here as possible. If the refresh count is specific to broken sites, I'd be tempted to move everything there too.

Reply:
Makes sense to me, will do.